src/server.rs

```diff
@@ -1413,16 +1471,26 @@ impl ServerRunner {
     ///
     /// This fails when the server cannot be started.
     pub fn run(self) -> Result<()> {
+        let Self {
+            server,
+            access_control_sentinel,
+        } = self;
+
         let status_code = ua::StatusCode::new(unsafe {
             UA_Server_runUntilInterrupt(
                 // SAFETY: Cast to `mut` pointer. Function is not marked `UA_THREADSAFE` but we make
                 // sure that it can only be invoked a single time (ownership of `ServerRunner`). The
                 // examples in `open62541` demonstrate that running the server in its own thread and
                 // interacting with it as we do through `Server` is okay.
-                self.0.as_ptr().cast_mut(),
+                server.as_ptr().cast_mut(),
             )
         });
-        Error::verify_good(&status_code)
+        Error::verify_good(&status_code)?;
+
+        // Only when the server has finished shutting down, we are allowed to drop sentinel values.
```

Comment:
This explicit drop has not effect, because it happens implicitly. Not only when returning with `Ok` but also when returning with a error one line above.

I consider adding it only for the `Ok` path confusing and misleading.